### PR TITLE
Shield re-work

### PR DESCRIPTION
**Pull request by Qustinnus**
TODO: get altangy or some other engineering nerd to test power needs for shields

TODO: have combat balance nerds look into how OP the shields are now and change them accordingly

TODO: Get aerwhisp to PR map changes to my PR because i suck at mapping

this gives shields new sprites, doesnt make a single hit destroy the shields, gives different weapons different capabilities against shields (with some being fully deflected)

we can now also add shield upgrades which I might add after this PR

image

image

https://i.gyazo.com/49559ab6aa0120fba72810b59a4aaecb.mp4

credits to polaris for some shield code

### CODE REVIEW

File: code/controllers/subsystem/ships.dm [hunk 1]
```diff
@@ -164,48 +164,52 @@ SUBSYSTEM_DEF(ship)
...
-				if(W.attack_data.shield_bust)
-					broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name], which pierced the shield and hit! Hit location: [target.loc].</span>",error_sound,S) //so the message doesn't get there early
-				else
-					broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit location: [target.loc].</span>",error_sound,S) //so the message doesn't get there early
-				for(var/mob/living/carbon/human/M in GLOB.player_list)
-					if(!istype(M.loc.loc, /area/shuttle/ftl))
-						continue
-					var/dist = get_dist(M.loc, target.loc)
-					shake_camera(M, dist > 20 ? 3 : 5, dist > 20 ? 1 : 3)
+			SSstarmap.ftl_sound('sound/weapons/ship_hit_shields.ogg')
```
Comment (TMTIME) [inline code/controllers/subsystem/ships.dm]: shield_hit_sound

**Comment by Ecolli** [comment]
Cant wait to see this go live

File: code/controllers/subsystem/ships.dm (hunk 1 above)
Comment (Qustinnus) [reply code/controllers/subsystem/ships.dm]: oh woops i forgot to upload the new sound I had

Comment (Qustinnus) [reply code/controllers/subsystem/ships.dm]: wait no u r a liar

**Pull request merged**